### ISSUE TITLE
OpticDiscSeg return label as path string rather than array when mode is 'test'

### ISSUE THREAD

**Issue by franva** [PaddlePaddle/PaddleSeg#958]
```
train_transforms = [
    #补全  
    T.RandomHorizontalFlip(),
    T.RandomRotation(max_rotation=10),
    T.RandomBlur(),
    T.RandomDistort(),
    T.RandomScaleAspect(min_scale=0.7, aspect_ratio=0.30),
    T.Resize(target_size=(512,512)),
    T.Normalize()
]
val_transforms = [
    #补全
    T.Resize(target_size=(512,512)),
    T.Normalize()
]

test_transforms = [
    # 补全
    T.Resize(target_size=(512, 512)),
    T.Normalize()
]

dataset_root = '/home/aistudio/work/newdata'
train_path  = '/home/aistudio/work/newdata/train_list.txt'
val_path  = '/home/aistudio/work/newdata/val_list.txt'
test_path  = '/home/aistudio/work/newdata/test_list.txt'
# 构建训练集
train_dataset = OpticDiscSeg(
    #补全
    dataset_root=dataset_root,
    transforms = train_transforms,
    mode='train')

#验证集
val_dataset = OpticDiscSeg(
    #补全
    dataset_root=dataset_root,
    transforms=val_transforms,
    mode='val')

#测试集
test_dataset = OpticDiscSeg(
    # 补全
    dataset_root=dataset_root,
    transforms=test_transforms,
    mode='test'
    )
```
```
timg, tlabel = test_dataset[131]
# type(timg), type(tlabel), type(label)
print('train label is: {}'.format(label))
print('test label is: {}'.format(tlabel))
```
_The output is:_

**train label is: [[255 255 255 ... 255 255 255]
 [255 255 255 ... 255 255 255]
 [255 255 255 ... 255 255 255]
 ...
 [  0   0   0 ... 255 255 255]
 [  0   0   0 ... 255 255 255]
 [  0   0   0 ... 255 255 255]]
test label is: /home/aistudio/work/newdata/origin/3352.jpg**

See above, the train label is an array, but the test label is a path string.
This is an inconsistent behavior, not sure is this intended or not?

**Comment by michaelowenliu** [comment]
@franva Hi, yes, it is an issue introduced by an old version, which is for prediction. In the current version, we designed a specific `predict` method for the purpose. The `test` mode is deprecated and will be removed in a future version. Thanks a lot!
